ksql-engine/src/test/java/io/confluent/ksql/topic/DefaultTopicInjectorTest.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.topic;
+
+import static io.confluent.ksql.topic.TopicInjectorFixture.hasTopicInfo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.hamcrest.Matchers.is;
+
+import com.google.common.collect.Lists;
+import io.confluent.ksql.ddl.DdlConfig;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.tree.CreateStreamAsSelect;
+import io.confluent.ksql.parser.tree.StringLiteral;
+import io.confluent.ksql.topic.TopicInjectorFixture.Inject;
+import io.confluent.ksql.topic.TopicInjectorFixture.Type;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Enclosed.class)
+public class DefaultTopicInjectorTest {
+
+  public static class Tests {
+    @Rule
+    public final TopicInjectorFixture fixture = new TopicInjectorFixture();
+
+    @Rule
+    public final ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void shouldInferTopicName() {
+      // Given:
+      fixture.givenStatement(
+          "CREATE STREAM sink WITH (value_format='JSON') AS SELECT * FROM source;"
+      );
+
+      // When:
+      final PreparedStatement<CreateStreamAsSelect> csas = fixture.inject();
```

Comment:
It's a bit weird that this unit test tests a test fixture, not the implementation directly.

Reply:
It's because of the `Enclosed` class - it doesn't let you share state.